src/libraries/Microsoft.Extensions.Options/gen/Parser.cs

```diff
@@ -0,0 +1,664 @@
+                        {
+                            Diag(DiagDescriptors.PotentiallyMissingEnumerableValidation, member.GetLocation(), enumeratedType.Name, member.Name);
+                        }
+                    }
+                }
+            }
+
+            if (validationAttrs.Count > 0 || transValidatorTypeName is not null || enumerationValidatorTypeName is not null)
+            {
+                return new(
+                    member.Name,
+                    validationAttrs,
+                    transValidatorTypeName,
+                    transValidatorIsSynthetic,
+                    enumerationValidatorTypeName,
+                    enumerationValidatorIsSynthetic,
+                    memberType.IsReferenceType || memberType.NullableAnnotation == NullableAnnotation.Annotated,
+                    memberType.IsValueType,
+                    enumeratedIsNullable,
+                    enumeratedIsValueType,
+                    enumeratedMayBeNull);
+            }
+
+            return null;
+        }
+
+        private string? AddSynthesizedValidator(ITypeSymbol modelType, ISymbol member)
+        {
+            var mt = modelType.WithNullableAnnotation(NullableAnnotation.None);
+            if (mt.OriginalDefinition.SpecialType == SpecialType.System_Nullable_T)
+            {
+                // extract the T from a Nullable<T>
+                mt = ((INamedTypeSymbol)mt).TypeArguments[0];
+            }
+
+            if (_synthesizedValidators.TryGetValue(mt, out var validator))
+            {
+                return "global::" + validator.Namespace + "." + validator.Name;
+            }
+
+            var membersToValidate = GetMembersToValidate(mt, true);
+            if (membersToValidate.Count == 0)
+            {
+                // this type lacks any eligible members
+                Diag(DiagDescriptors.NoEligibleMember, member.GetLocation(), mt.ToString(), member.ToString());
+                return null;
+            }
+
+            var model = new ValidatedModel(
+                GetFQN(mt),
+                mt.Name,
+                false,
+                membersToValidate);
+
+            var validatorTypeName = "__" + mt.Name + "Validator__";
+
+            var result = new ValidatorType(
+                mt.ContainingNamespace.IsGlobalNamespace ? string.Empty : mt.ContainingNamespace.ToString(),
+                validatorTypeName,
+                validatorTypeName,
+                "class",
+                new List<string>(),
+                true,
+                new[] { model });
+
+            _synthesizedValidators[mt] = result;
+            return "global::" + (result.Namespace.Length > 0 ? result.Namespace + "." + result.Name : result.Name);
+        }
+
+        private bool DerivesFrom(ITypeSymbol source, ITypeSymbol dest)
+        {
+            var conversion = _compilation.ClassifyConversion(source, dest);
+            return conversion.IsReference && conversion.IsImplicit;
```

Comment:
nit - wondering if this method should have a different name; does this result comprehensively indicate that there's an inheritance relationship here?

Reply:
I changed the name to `ConvertTo`.